x-pack/filebeat/input/netflow/decoder/v9/v9.go

```diff
@@ -61,12 +68,16 @@ func (p *NetflowV9Protocol) Start() error {
 	if p.timeout != time.Duration(0) {
 		go p.Session.CleanupLoop(p.timeout, p.done)
 	}
+
+	p.cache.start(p.done, 30*time.Second, 10*time.Second)
+
 	return nil
 }
 
 func (p *NetflowV9Protocol) Stop() error {
 	if p.done != nil {
 		close(p.done)
+		p.cache.stop()
```

Comment:
nothing prevents double close on the channel here

example, added extra stop in the existing test:
```
func TestNetflowProtocol_New(t *testing.T) {
	proto := New(config.Defaults())

	assert.Nil(t, proto.Start())
	assert.Equal(t, uint16(9), proto.Version())
	assert.Nil(t, proto.Stop())
	assert.Nil(t, proto.Stop())
}
```
results in:
```
panic: close of closed channel [recovered]
	panic: close of closed channel
```

Reply:
hmm this was inherited by the pre-existing netflow code but now with the context.Context Done utilisation this shouldn't happen. feel free to unresolve if you spot anything else